### ISSUE TITLE
aria roles on navigation items [accessibility]

### ISSUE THREAD

**Issue by joahua** [jnprr/blazor-breadcrumb-navigation#3]
Hi there,

Thanks for putting this component together!

I was wondering if you'd consider adding aria roles to the breadcrumbs. These are helpful for users of assistive technologies who might rely on them to navigate pages. https://www.w3.org/TR/wai-aria-practices-1.1/examples/breadcrumb/index.html has a reference example.

There are two attributes needed to achieve this.
1. The outermost container
2. The last child item

The outermost container should have an attribute of `aria-label="Breadcrumb"` - though perhaps this could be left as an exercise for the user, given it's possible to just wrap the breadcrumb ul in a <nav> element with that label? If this was something you were open to adding directly, perhaps best to wrap the <ul> in `src/Supercode.Blazor.BreadcrumbNavigation/BreadcrumbNavigation.razor` with `<nav aria-label="Breadcrumb">`, as guidance is that the old `role="navigation"` attributes are no longer required? Otherwise perhaps this is something that could just be adjusted in the docs under https://github.com/cschulzsuper/blazor-breadcrumb-navigation#use-breadcrumbnavigation-in-mainlayout to give an example?

More importantly - as it can't be implemented without support in this component - the current/last item, whether a link or text, should have an attribute of `aria-current="page"`. I think this needs a new `builder.AddAttribute` for both `RenderFragmentNavLink` and `RenderFragmentText` in `/src/Supercode.Blazor.BreadcrumbNavigation/Services/Breadcrumb.cs` - but I'm not sure what the best way is to let them know when a fragment is the last child?

**Comment by cschulzsuper** [comment]
Thanks Josh, for the thorough explanation of the issue. 

It would make sense that the component will take care of `aria-label="Breadcrumb"` and `aria-current="page"`. If I find time in the next couple of days, I'll try to evaluate this.

Detection of the last child will probably be the hardest part.

**Comment by cschulzsuper** [comment]
With small change on my side, it would be possible to extend the `BreadcrumbNavigation` and the `Breadcrumb` base class with `AdditionalAttributes`. The change will allow you to add `aria-label="Breadcrumb"` and` aria-current="page"`. It will also allow other custom attributes as well  

The usage of the components would change a bit.

``` html
    <BreadcrumbNavigation aria-label="Breadcrumb" />
```

``` csharp
    BreadcrumbService
        .Set<IndexBreadcrumb>()
        .Set<CounterBreadcrumb>(new Dictionary<string, object>
        {
            ["counter"] = currentCount,
            ["aria-current"] = "page"
        });
```
All parameters, which are not used inside of the breadcrumb, are added automatically to the link or the text. 

I like this solution, as I do not need to add the aria attributes myself.

Please let me know if that solution would fit your needs.

**Comment by joahua** [comment]
Thanks, that seems great to me - sorry for my slow response!

That'd also be handy for hooking [test IDs](https://playwright.dev/docs/selectors/#id-data-testid-data-test-id-data-test-selectors) or attributes for analytics purposes etc - quite tidy! 👍

**Comment by cschulzsuper** [comment]
I implemented a first draft for this.

- [issue/3/additional-attributes](https://github.com/cschulzsuper/blazor-breadcrumb-navigation/tree/issue/3/additional-attributes)

I made some adjustments compared to my initial idea. It is now possible to add additional attributes in the `Breadcrumb` classes. This gives developers the flexibility to add any kind of attribute not only to the _Link_ or _Text_ but also the _Icons_.

You have now to possible ways to set the `aria-current="page"`.

1. Duplicate your breadcrumbs, add the aria attribute to the duplicate and use the duplicate when you need it to be the last breadcrumb. I use this approach in the examples. 

``` csharp
public class IndexBreadcrumb : RootBreadcrumb
{
    public override void Configure(BreadcrumbBuilder builder)
        => builder
            .LeftIcon("oi oi-home")
            .Link("Home", string.Empty);
}

public class IndexBreadcrumbLast : RootBreadcrumb
{
    public override void Configure(BreadcrumbBuilder builder)
        => builder
            .LeftIcon("oi oi-home")
            .Link("Home", string.Empty,
                new KeyValuePair<string, object>("aria-current", "page"));
}
```

2. Add a `[Parameter]` to your breadcrumb and, depeding on the value, add the _aria_ attribute or not.

``` csharp
public class IndexBreadcrumb : RootBreadcrumb
{
    [Parameter]
    public bool Last { get; set; }

    public override void Configure(BreadcrumbBuilder builder)
    {
        builder.LeftIcon("oi oi-home");

        if (Last) 
        {
            builder.Link("Home", string.Empty, new KeyValuePair<string, object>("aria-current", "page"));
        }
        else 
        {
            builder.Link("Home", string.Empty);
        }
    }
}

**Comment by cschulzsuper** [comment]
I published a new release that includes the changes. It should be avaiable on [NuGet](https://www.nuget.org/packages/Supercode.Blazor.BreadcrumbNavigation/).
